Rakefile

```diff
@@ -87,6 +87,24 @@ def converted_history(markdown)
           normalize_bullets(markdown)))))
 end
 
+def siteify_file(file, front_matter = {})
+  abort "You seem to have misplaced your #{file} file. I can haz?" unless File.exists?(file)
+  contents = File.read(file)
+  title = contents.match(/\A# (.*)$/)[1]
+  output_file = file.sub(/\.markdown\z/, ".md").downcase
+  slug = File.basename(output_file, ".md")
+  front_matter = front_matter.merge({
+    "title"     => title,
+    "layout"    => "docs",
+    "permalink" => "/docs/#{slug}/",
+    "note"      => "This file is autogenerated. Edit /#{file} instead."
+  })
+  contents.gsub!(/\A# #{title}\n\n?/, "")
+  contents = converted_history(contents) if output_file == "history.md"
```

Comment:
> I would prefer to leave this out... This is a special case I'd rather avoid.
 
These two statements feel contradictory to me. Do you have any suggestions on how to convert the history otherwise? As a pre-process step? Leave the history processing as it is?